Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -258,7 +258,10 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
           // It's not necessary to restate the postcondition itself if the user is already hovering it
           // however, nested postconditions should be displayed
           if (errorToken is BoogieRangeToken rangeToken && !hoveringPostcondition) {
-            deltaInformation += "  \n" + CouldProveOrNotPrefix + ideState.TextDocumentItem.Text.Substring(rangeToken.StartToken.pos,
+            var originalText =
+              ideState.Uri == rangeToken.Filename ? ideState.TextDocumentItem.Text :
+              File.ReadAllText(rangeToken.ActualFilename);
```

Comment:
It's probably worth waiting for a future PR, so that this issue can be fixed quickly, but it would be nice in the long run to cache the text of files already looked at so that the server doesn't need to re-read the file every time someone hovers.

Reply:
I agree. looks like it's going to be more complex than I thought though..